File: food_order_app/lib/domain/food/entities/food_entity.dart
class FoodEntity {
  final String id;
  final String name;
  final String description;
  final double price;
  final String categoryId;
  final String categoryName;
  final String imageUrl;
  final bool isAvailable;

  const FoodEntity({
    required this.id,
    required this.name,
    this.description = '',
    required this.price,
    this.categoryId = '',
    this.categoryName = '',
    this.imageUrl = '',
    this.isAvailable = true,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'categoryId': categoryId,
      'categoryName': categoryName,
      'imageUrl': imageUrl,
      'isAvailable': isAvailable,
    };
  }

  factory FoodEntity.fromJson(Map<String, dynamic> json) {
    return FoodEntity(
      id: (json['id'] ?? '').toString(),
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] is num) ? (json['price'] as num).toDouble() : 0.0,
      categoryId: json['categoryId'] ?? '',
      categoryName: json['categoryName'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      isAvailable: json['isAvailable'] ?? true,
    );
  }
}

File: food_order_web/src/domain/foods/entities/food_entity.js
export class FoodEntity {
  constructor({
    id = '',
    name = '',
    description = '',
    price = 0,
    categoryId = '',
    categoryName = '',
    imageUrl = '',
    isAvailable = true,
    rating = 4.8,
  } = {}) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = Number(price) || 0;
    this.categoryId = categoryId;
    this.categoryName = categoryName;
    this.imageUrl = imageUrl;
    this.isAvailable = Boolean(isAvailable);
    this.rating = Number(rating) || 4.8;
  }

  get formattedPriceUSD() {
    return `$${this.price.toFixed(2)}`;
  }

  get formattedPriceKHR() {
    const khr = Math.round(this.price * 4100);
    return `${khr.toLocaleString()} ៛`;
  }
}

export class CategoryEntity {
  constructor({
    id = '',
    name = '',
    icon = '🍽️',
    description = '',
    order = 0,
  } = {}) {
    this.id = id;
    this.name = name;
    this.icon = icon;
    this.description = description;
    this.order = Number(order) || 0;
  }
}
